assets/dev/js/editor/controls/behaviors/scrubbing.js

```diff
@@ -0,0 +1,160 @@
+const SCRUB_REGULAR = 'UPDATE-VALUE';
+const SCRUB_ENHANCED = 'UPDATE-VALUE-ENHANCED';
+const SKIP_SCRUB = 'SKIP-UPDATE-VALUE';
+
+export default class Scrubbing extends Marionette.Behavior {
+	checkIntentTimeout = null;
+
+	skipperCount = 0;
+
+	constructor( ...args ) {
+		super( ...args );
+
+		const userOptions = this.getOption( 'scrubSettings' ) || {};
+
+		this.scrubSettings = {
+			intentTime: 600,
+			skipperSteps: 10,
+			enhancedNumber: 10,
+			scrubbingActiveClass: 'e-scrubbing--active',
+			scrubbingOverClass: 'e-scrubbing-over',
+			...userOptions,
+		};
+	}
+
+	ui() {
+		return {
+			input: 'input[type=number]',
+			label: 'label',
+		};
+	}
+
+	events() {
+		return {
+			'mousedown @ui.input': 'onMouseDownInput',
+			'mousedown @ui.label': 'onMouseDownLabel',
+			'mouseenter @ui.label': 'onMouseEnterLabel',
+			'mouseleave @ui.label': 'onMouseLeaveLabel',
+		};
+	}
+
+	scrub( input, movementEvent ) {
+		const movementType = this.getMovementType( movementEvent );
+
+		if ( SKIP_SCRUB === movementType ) {
+			return;
+		}
+
+		switch ( movementType ) {
+			case SCRUB_REGULAR:
+				input.value = +input.value + movementEvent.movementX;
+				break;
+
+			case SCRUB_ENHANCED:
+				input.value = +input.value + ( movementEvent.movementX * this.scrubSettings.enhancedNumber );
+				break;
+
+			default:
+				break;
+		}
+
+		// Fire an input event so other behaviors/validators can handle the new input
+		input.dispatchEvent( new Event( 'input', { bubbles: true } ) );
+	}
+
+	getMovementType( movementEvent ) {
+		if ( movementEvent.altKey ) {
+			this.skipperCount++;
+
+			// When ALT key is pressed, skipping x times before updating input value.
+			// 'skipperSteps' defines the exact times to skip, can be changed to any number.
+			if ( this.skipperCount <= this.scrubSettings.skipperSteps ) {
+				return SKIP_SCRUB;
+			}
+
+			this.skipperCount = 0;
+
+			return SCRUB_REGULAR;
+		}
+
+		return ( movementEvent.ctrlKey || movementEvent.metaKey ) ? SCRUB_ENHANCED : SCRUB_REGULAR;
+	}
+
+	isInputValidForScrubbing( input ) {
+		return input && ! input.disabled && 'number' === input.type;
+	}
+
+	onMouseDownInput( e ) {
+		const input = e.target;
+
+		if ( ! this.isInputValidForScrubbing( input ) ) {
+			return;
+		}
+
+		const trackMovement = ( movementEvent ) => {
+			this.scrub( input, movementEvent );
+		};
+
+		// For input, scrubbing effect works only after X time the mouse is down.
+		const checkIntentTimeout = setTimeout( () => {
+			clearTimeout( checkIntentTimeout );
+			document.addEventListener( 'mousemove', trackMovement );
+			$e.uiStates.set( 'document/scrubbing-mode', 'on' );
+			input.classList.add( this.scrubSettings.scrubbingActiveClass );
```

Comment:
do me a favor and add some spaces between the lines, based on the "subject" of the code block

Reply:
The php lint displays an error, need to configure better.

Reply:
Done.